### PR TITLE
Improves Donkcode

### PR DESCRIPTION
**Pull request by Geevies**
* Donk Pockets can now be heated properly in a microwave, they gain additional nutriment, animal protein and healing juice.
* Sin-Pockets now heat properly when used in hand, heats instantly, and gives slightly more hyperzine and synaptizine than before.

A curious side-effect I encountered in testing. You have to chew a lot more with the added reagents. But the benefit outweighs the consequence. Also, a traitor ravenously chewing a sin-pocket with a revolver in the other hand, bleeding profusely from everywhere, makes me giggle like a child.

Resolves https://github.com/Aurorastation/Aurora.3/issues/6802

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Comment by alsoandanswer** [comment]
Do they cool back down?

**Comment by Geevies** [comment]
No.

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 1]
```diff
@@ -998,61 +998,53 @@
...
 
-/obj/item/weapon/reagent_containers/food/snacks/donkpocket/proc/heat()
-	warm = 1
-	for(var/reagent in heated_reagents)
-		reagents.add_reagent(reagent, heated_reagents[reagent])
-	bitesize = 6
-	name = "Warm " + name
-	cooltime()
+/obj/item/weapon/reagent_containers/food/snacks/donkpocket/warm
+	name = "Warm Donk-pocket"
```
Comment (NonQueueingMatt) [inline code/modules/reagents/reagent_containers/food/snacks.dm]:
```suggestion
	name = "warm Donk-pocket"
```

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 2]
```diff
@@ -998,61 +998,53 @@
...
-	..()
-	if (!warm)
-		heat()
+/obj/item/weapon/reagent_containers/food/snacks/donkpocket/sinpocket/attack_self(mob/user)
+	if(has_been_heated)
+		to_chat(user, "<span class='notice'>The heating chemicals have already been spent.</span>")
+		return
+	has_been_heated = TRUE
+	user.visible_message("<span class='notice'>[user] crushes \the [src] package.</span>", "You crush \the [src] package and feel it rapidly heat up.")
+	name = "\improper Warm Sin-pocket"
```
Comment (NonQueueingMatt) [inline code/modules/reagents/reagent_containers/food/snacks.dm]:
```suggestion
	name = "warm Sin-pocket"
```

**Comment by JohnWildkins** [comment]
If you wanted, I feel like you could add a timer on the warm donk's creation to turn them back into normal donks if they aren't eaten in like, five or ten minutes.

**Comment by Geevies** [comment]
I don't intend to add fluff code to let them heat up and cool them down again. It's precisely this that caused them to bug out in the past.

**Comment by alsoandanswer** [comment]
Still, would be kind of weird for it to say "warm" donk pockets even though it's left out for like half an hour. Maybe make it like one of the things with autoheat on click or just rename the post microwave version to "cooked" donk pockets

Hm, here's an idea. Microwaving or putting them in an oven will give different nutritional value.

Like, "baked" donk pockets and "microwaved" donk pockets.

Anything's better than thermodynamically violating pizza pockets.

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 3]
```diff
@@ -998,61 +998,53 @@
...
 
-/obj/item/weapon/reagent_containers/food/snacks/donkpocket/proc/heat()
-	warm = 1
-	for(var/reagent in heated_reagents)
-		reagents.add_reagent(reagent, heated_reagents[reagent])
-	bitesize = 6
-	name = "Warm " + name
-	cooltime()
+/obj/item/weapon/reagent_containers/food/snacks/donkpocket/warm
+	name = "warm Donk-pocket"
```
Comment (Geevies) [inline code/modules/reagents/reagent_containers/food/snacks.dm]:
```suggestion
	name = "baked Donk-pocket"
```

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 4]
```diff
@@ -998,61 +998,53 @@
...
+	desc = "The warm, reheatable food of choice for the seasoned spaceman."
```
Comment (Geevies) [inline code/modules/reagents/reagent_containers/food/snacks.dm]:
```suggestion
	desc = "The baked, reheatable food of choice for the seasoned spaceman."
```

**Comment by alsoandanswer** [comment]
Do it for sin-pockets too and used cooked instead of baked and we all gucci

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 5]
```diff
@@ -998,61 +998,53 @@
...
-	..()
-	if (!warm)
-		heat()
+/obj/item/weapon/reagent_containers/food/snacks/donkpocket/sinpocket/attack_self(mob/user)
+	if(has_been_heated)
+		to_chat(user, "<span class='notice'>The heating chemicals have already been spent.</span>")
+		return
+	has_been_heated = TRUE
+	user.visible_message("<span class='notice'>[user] crushes \the [src] package.</span>", "You crush \the [src] package and feel it rapidly heat up.")
+	name = "warm Sin-pocket"
```
Comment (Geevies) [inline code/modules/reagents/reagent_containers/food/snacks.dm]:
```suggestion
	name = "cooked Sin-pocket"
```

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 6]
```diff
@@ -998,61 +998,53 @@
...
 
-/obj/item/weapon/reagent_containers/food/snacks/donkpocket/proc/heat()
-	warm = 1
-	for(var/reagent in heated_reagents)
-		reagents.add_reagent(reagent, heated_reagents[reagent])
-	bitesize = 6
-	name = "Warm " + name
-	cooltime()
+/obj/item/weapon/reagent_containers/food/snacks/donkpocket/warm
+	name = "baked Donk-pocket"
```
Comment (alsoandanswer) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: cooked here too

Comment (Geevies) [inline code/modules/reagents/reagent_containers/food/snacks.dm]:
```suggestion
	name = "cooked Donk-pocket"
```

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 7]
```diff
@@ -998,61 +998,53 @@
...
+	desc = "The baked, reheatable food of choice for the seasoned spaceman."
```
Comment (Geevies) [inline code/modules/reagents/reagent_containers/food/snacks.dm]:
```suggestion
	desc = "The cooked, reheatable food of choice for the seasoned spaceman."
```

**Pull request merged**